### PR TITLE
asm: aarch64: Integrate aarch64 assembly: src/arm/64/ipred16.S 

### PR DESCRIPTION
**Pull request by vibhoothiiaanand**
This commit,
* Integrates the following functions

- [x] rav1e_ipred_dc_16bpc_neon,
- [x] rav1e_ipred_dc_128_16bpc_neon,
- [x] rav1e_ipred_dc_left_16bpc_neon,
- [x] rav1e_ipred_dc_top_16bpc_neon,
- [x] rav1e_ipred_v_16bpc_neon,
- [x] rav1e_ipred_h_16bpc_neon,
- [x] rav1e_ipred_smooth_16bpc_neon,
- [x] rav1e_ipred_smooth_v_16bpc_neon,
- [x] rav1e_ipred_smooth_h_16bpc_neon,
- [x] rav1e_ipred_paeth_16bpc_neon,
- [x] rav1e_ipred_cfl_16bpc_neon,
- [x] rav1e_ipred_cfl_128_16bpc_neon,
- [x] rav1e_ipred_cfl_left_16bpc_neon,
- [x] rav1e_ipred_cfl_top_16bpc_neon,

* Make the condition idiomatic choosing between HBD and LBD

* Fixes the calling convention for LBD function which is not having two missing parameters for angular_ipred_fns

* Moved the tests for pred_matches_u8 for AVX2 to the shared module, to share same tests b/w AVX2 and NEON.

Advances #2376

### CODE REVIEW

File: src/asm/aarch64/predict.rs [hunk 1]
```diff
@@ -80,64 +128,187 @@ pub fn dispatch_predict_intra<T: Pixel>(
...
+              angle,
+              0,
+              0,
+              (1 << bit_depth) - 1,
+            );
+          }
+          PredictionMode::PAETH_PRED => {
+            rav1e_ipred_paeth_16bpc_neon(
+              dst_hbd_ptr,
+              T::to_asm_stride(stride as usize),
```
Comment (lu-zero) [inline src/asm/aarch64/predict.rs:281]: why not keeping it at the top instead?

File: src/asm/shared/predict.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+#[cfg(test)]
+mod test {
+
```
Comment (lu-zero) [inline src/asm/shared/predict.rs:12]: I'd remove the empty line

File: src/asm/aarch64/predict.rs (hunk 1 above)
Comment (vibhoothiiaanand) [reply src/asm/aarch64/predict.rs:281]: indeed that was a nice catch ^^;

File: src/asm/aarch64/predict.rs [hunk 3]
```diff
@@ -80,50 +128,60 @@ pub fn dispatch_predict_intra<T: Pixel>(
     );
   };
 
-  if size_of::<T>() != 1 {
-    return call_rust(dst);
+  if cpu < CpuFeatureLevel::NEON {
+    call_rust(dst)
```
Comment (barrbrain) [inline src/asm/aarch64/predict.rs:132]: Missing `return` here.

**Comment by vibhoothiiaanand** [comment]
| Before #2501 | After #2501 |  
-- | -- | -- | --
FPS | 0.220 | 0.230 |4.34%
Tiime | 454 | 435 | 4.18%

On Chimera 10 bit 100 frames on honeycomb LX2K with verbose on stable

Command: `time cargo run --release ~/chimera-10b-1080p-400.y4m -l 100 -o /dev/null --tiles 15 --verbose`

**Pull request merged**